src/boot/efi/boot.c

```diff
@@ -665,58 +718,64 @@ static bool menu_run(
                         /* account for padding+status */
                         visible_max = y_max - 2;
 
+                        visible_entries = entries_count(config, visible_group);
+
                         /* Drawing entries starts at idx_first until idx_last. We want to make
                         * sure that idx_highlight is centered, but not if we are close to the
                         * beginning/end of the entry list. Otherwise we would have a half-empty
                         * screen. */
-                        if (config->n_entries <= visible_max || idx_highlight <= visible_max / 2)
+                        if (visible_entries <= visible_max || idx_highlight <= visible_max / 2)
                                 idx_first = 0;
-                        else if (idx_highlight >= config->n_entries - (visible_max / 2))
-                                idx_first = config->n_entries - visible_max;
+                        else if (idx_highlight >= visible_entries - (visible_max / 2))
+                                idx_first = visible_entries - visible_max;
                         else
                                 idx_first = idx_highlight - (visible_max / 2);
                         idx_last = idx_first + visible_max - 1;
 
                         /* length of the longest entry */
                         line_width = 0;
-                        for (size_t i = 0; i < config->n_entries; i++)
-                                line_width = MAX(line_width, strlen16(config->entries[i]->title_show));
+                        for (size_t i = 0; i < config->n_entries; i++){
+                                if(entry_is_visible(config->entries[i], visible_group))
+                                        line_width = MAX(line_width, strlen16(config->entries[i]->title_show));
+                        }
```

Comment:
Coding style: no {} when not needed, as space between `if` and `(`.

Reply:
This seems to be outdated now :)
I tried to check everything, but it might be that I missed parts